### PR TITLE
1.4/1126 write back end submission process 2

### PR DESCRIPTION
**Pull request by charlesstrange2**
Basic functionality in play.
https://trello.com/c/QoGCdPWa/1126-write-back-end-submission-process-update-bundle-dispersement
Could do with some further validation guards on letting users assign wacky values to the PUT update route. 
Also requires tests, see test card.
https://trello.com/c/gRCtBfV7/1138-write-tests-to-check-voucher-disbersement

### CODE REVIEW

File: app/Http/Controllers/Store/BundleController.php [hunk 1]
```diff
@@ -162,10 +181,47 @@ function ($value) {
...
+
+                // Find and add the centre
+                $centre = Centre::findOrFail($inputs['collected_at']);
+                $bundle->disbursingCentre()->associate($centre);
+
+                // Add the current user as disbursingUser.
+                $bundle->disbursingUser()->associate(Auth::user());
+
+                // Store it.
+                $bundle->save();
```
Comment (RpprRoger) [inline app/Http/Controllers/Store/BundleController.php]: Large try/catches can hide errors which you don't expect. The expected errors _only_ occur on line 198 and 201. From what I can tell?

File: app/Http/Controllers/Store/BundleController.php [hunk 2]
```diff
@@ -188,27 +244,33 @@ public function removeVoucherFromCurrentBundle(Registration $registration, Vouch
...
      */
-    public function redirectAfterRequest($errors, $registration)
+    public function redirectAfterRequest($errors, $successRoute, $failRoute)
     {
-        $route = route('store.registration.voucher-manager', ['registration' => $registration->id]);
         if (!empty($errors)) {
             // Assemble messages
             $messages = [];
             foreach ($errors as $type => $values) {
                 switch ($type) {
```
Comment (RpprRoger) [inline app/Http/Controllers/Store/BundleController.php]: so `errors["transactions"] == false` will still trigger this error message? That's a tad confusing.

File: app/Http/Controllers/Store/BundleController.php [hunk 3]
```diff
@@ -162,10 +181,47 @@ function ($value) {
             $errors[] = $bundle->syncVouchers($voucherCodes);
         }
 
-        // TODO : disbursing bundle code to go here.
+        // Check we have values on our inputs; This should have been covered in validation...
```
Comment (RpprRoger) [inline app/Http/Controllers/Store/BundleController.php]: > This should have been covered in validation...

:)

File: app/Http/Controllers/Store/BundleController.php [hunk 4]
```diff
@@ -162,10 +181,47 @@ function ($value) {
...
+        if (isset($inputs['collected_at']) &&
+            isset($inputs['collected_by']) &&
+            isset($inputs['collected_on'])
+        ) {
+            // Add the date;
+            $bundle->disbursed_at = Carbon::createFromFormat(
+                'Y-m-d',
+                $inputs['collected_on']
```
Comment (RpprRoger) [inline app/Http/Controllers/Store/BundleController.php]: Disbursed is the same as collected? I presume that makes sense from the data's pov but it's a bit confusing to read.

**Comment by RpprRoger** [comment]
This code will work looks logical. I just have minor feedback

File: app/Http/Controllers/Store/BundleController.php (hunk 3 above)
Comment (charlesstrange2) [reply app/Http/Controllers/Store/BundleController.php]: yeah. It should be, but i also thought i'd check they were actually filled here.

**Pull request merged**